conf/log.go

```diff
@@ -105,14 +124,96 @@ func (l *Log) CheckDefault() {
 	}
 }
 
+// Open open the log file
+func (l *Log) Open() {
+	// using stdout if no log file
+	if l.File == "" {
+		l.Writter = os.Stdout
+		return
+	}
+	// using lumberjack if self rotate
+	if l.SelfRotate == true {
+		log.Debugf("[Log] Self Rotate log file %s", l.File)
+		l.Writter = &lumberjack.Logger{
+			Filename:   l.File,
+			MaxSize:    l.MaxSize, // megabytes
+			MaxBackups: l.MaxBackups,
+			MaxAge:     l.MaxAge, //days
+			Compress:   l.Compress,
+		}
+		return
+	}
+	// using log file if not self rotate
+	f, err := os.OpenFile(l.File, os.O_APPEND|os.O_CREATE|os.O_RDWR, 0640)
+	if err != nil {
+		log.Warnf("[Log] Cannot open log file: %v", err)
+		log.Infoln("[Log] Using Standard Output as the log output...")
+		l.Writter = os.Stdout
+		return
+	}
+	l.Writter = f
+}
+
+// Close close the log file
+func (l *Log) Close() {
+	if l.Writter != nil {
+		if f, ok := l.Writter.(*os.File); ok {
+			f.Close()
+		}
+	}
+}
+
 // GetWriter return the log writer
-func (l *Log) GetWriter() *lumberjack.Logger {
-	return &lumberjack.Logger{
-		Filename:   l.File,
-		MaxSize:    l.MaxSize, // megabytes
-		MaxBackups: l.MaxBackups,
-		MaxAge:     l.MaxAge, //days
-		Compress:   l.Compress,
+func (l *Log) GetWriter() io.Writer {
+	if l.Writter == nil {
+		l.Open()
+	}
+	return (io.Writer)(l.Writter)
+}
+
+//Rotate rotate the log file
+func (l *Log) Rotate() {
+	if l.Writter != nil {
+		if lumberjackLogger, ok := l.Writter.(*lumberjack.Logger); ok {
+			// self rotate
+			if err := lumberjackLogger.Rotate(); err != nil {
+				log.Errorf("[Log] Rotate log file failed: %s", err)
+			}
+		} else if fileLogger, ok := l.Writter.(*os.File); ok {
```

Comment:
The `Writer` could be `os.Stdout`, we should not close it in that case.

Reply:
this check makes sure the Writer is `*os.File` type only.

Reply:
Type of https://pkg.go.dev/os#Stdout look like *os.File